src/commonMain/kotlin/arrow.exact/Exact.kt

```diff
@@ -0,0 +1,21 @@
+package arrow.exact
+
+import arrow.core.*
+
+interface Exact<A, out B> {
+
+  fun fromOrEither(value: A): Either<ExactError, B>
+
+  fun fromOrNull(value: A): B? {
+    return fromOrEither(value).orNull()
+  }
+
+  fun fromOrThrow(value: A): B {
+    return when (val result = fromOrEither(value)) {
+      is Either.Left -> throw IllegalArgumentException(result.value.message)
```

Comment:
We should probably create an `ArrowExactException` containing more information but that's not critical for the draft. I've defined an example for such error in #5 .

Reply:
@ILIYANGERMANOV You are right, will create a custom exception and extend it from `IllegalArgumentException` as @nomisRev has supposed. Indeed it will be useful to put there some extra information about the failed validation

But now I see that as a client I probably would want to define my own exception, maybe we can design it like in Result.getOrElse:

```
fromOrElse(onFailure: (kotlin.Throwable) -> R): R
```

Reply:
Perfect, I imagine it like that - custom exception extending `IllegalArgumentException` :+1: